include/aspect/particle/property/viscoplastic_strain_invariants.h

```diff
@@ -0,0 +1,135 @@
+/*
+ Copyright (C) 2015 - 2017 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file LICENSE.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect_particle_property_viscoplastic_strain_invariant_h
+#define _aspect_particle_property_viscoplastic_strain_invariant_h
+
+#include <aspect/particle/property/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/material_model/visco_plastic.h>
+
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      /**
+       * A class that calculates the finite strain invariant a particle has
+       * experienced and assigns it to the plastic, viscous, or total strain fields
+       * used in the visco_plastic material model. The implementation of this property
+       * is equivalent to the implementation for compositional fields that is located in
+       * the plugin <code>benchmarks/buiter_et_al_2008_jgr/plugin/finite_strain_invariant.cc</code>.
+       *
+       * @ingroup ParticleProperties
+       */
+      template <int dim>
+      class ViscoPlasticStrainInvariant : public Interface<dim>, public ::aspect::SimulatorAccess<dim>
+      {
+        public:
+          /**
+          * Constructor.
+          */
+          ViscoPlasticStrainInvariant ();
+
+          virtual void initialize ();
+
+
+          /**
+           * Initialization function. This function is called once at the
+           * creation of every particle for every property to initialize its
+           * value.
+           *
+           * @param [in] position The current particle position.
+           * @param [in,out] particle_properties The properties of the particle
+           * that is initialized within the call of this function. The purpose
+           * of this function should be to extend this vector by a number of
+           * properties.
+           */
```

Comment:
you do not need to copy this documentation. Either write a documentation that is specific for this plugin, or use the following:
```
/**
 * @copydoc aspect::Particle::Property::Interface::initialize_one_particle_property()
**/
```

The same applies to all the other functions below.